CMakeLists.txt

```diff
@@ -89,11 +89,8 @@ elseif ((ONNX_USE_PROTOBUF_SHARED_LIBS AND Protobuf_USE_STATIC_LIBS)
   message(FATAL_ERROR
     "ONNX_USE_PROTOBUF_SHARED_LIBS and Protobuf_USE_STATIC_LIBS must be opposites of each other.")
 endif()
-add_subdirectory(third_party/onnx)
-
-add_subdirectory(third_party/pybind11)
-add_subdirectory(third_party/rapidcheck)
 
+add_subdirectory(third_party)
```

Comment:
In my experience, the `third_party` directory should only contain the submodules and NOT any custom `cmake` files. Adding a `cmake` file at the `third_party` directory level makes this less readable and harder to maintain.

Reply:
I do not understand the reason the `third_party` directory should not contain a `cmake` file.  Other directories already contain `cmake` files (e.g. src/CMakefile.txt) and I do not find that any harder to maintain.

Reply:
> I'm not sure what "no warning policy" you are thinking. Once again, the proper way to fix warnings is to fix your source code, not to sweep the dirt under the carpet.

@gongsu832 I don't see it as our direct responsibility to fix warnings of the third_party directories. However, as they generate a lot of spurious code, it pollutes the output that shows up on our shells as we compile onnx-mlir. A results is that we pay less attentions to our own warnings, thus they tend to increase.

Thus I see this as a good effort to ensure we are more disciplined in our efforts of avoiding warnings.